rudder-core/src/main/scala/com/normation/rudder/services/policies/JavascriptEngine.scala

```diff
@@ -0,0 +1,614 @@
+
+  /*
+   * A sandboxed security manager, allowing only restricted
+   * set of operation for restricted thread.
+   * The RudderJsEngineThreadFactory take care of correctly initializing
+   * threads.
+   *
+   * BE CAREFULL - this sandbox won't prevent an attacker to get/do things.
+   * It's goal is to prevent a user to do obviously bad things on the servers,
+   * but more by mistake (because he didn't understood when the script is evaled,
+   * for example).
+   *
+   */
+  protected[policies] class SandboxSecurityManager extends SecurityManager {
+    //by default, we don't sand box. Thread will do it in their factory
+    val isSandboxed = {
+      val itl = new InheritableThreadLocal[Boolean]()
+      itl.set(false)
+      itl
+    }
+
+    //authorized / needed runtime permissions
+    private[this] val runtimePerms = Set(
+        "modifyThread"      //needed by thread factory to try to kill the thread
+      , "createClassLoader" //needed by rhino to run almost anything, like creating a var - else NPE
+      , "nashorn.createGlobal" //needed by Nashorn to do stuff
+      , "accessDeclaredMembers" //needed by Nashorn to do stuff
+      , "loadLibrary.sunec"     //needed by Rudder JS Lib
+      , "getProtectionDomain"
+    )
+    private[this] val reflectPerms = Set(
+        "suppressAccessChecks" //needed by rhino to run almost anything, like creating a var - else NPE
+    )
+    private[this] val securityPerms = Set( //needed by rudder js lib
+        //we are checking the start for them
+        "getProperty.security.provider"
+      , "putProviderProperty"
+      , "getProperty.securerandom.source"
+      , "getProperty.jdk.certpath.disabledAlgorithms"
+    )
+
+    private[this] val filePerms = Set( //we only authorize read access
+        "/dev/random"
+      , "/dev/urandom"
+    )
+
+
+    override def checkPermission(permission: Permission): Unit = {
+      if(isSandboxed.get) {
+        permission match {
+          case x: FilePermission     if( x.getActions == "read" && filePerms.contains(x.getName)    ) => // ok
+          case x: SecurityPermission if( securityPerms.exists( p => x.getName.startsWith(p) )       ) => // ok
+          case x: RuntimePermission  if( x.getName.startsWith("accessClassInPackage.jdk.nashorn")   ) => // ok
+          case x: RuntimePermission  if( x.getName.startsWith("accessClassInPackage.sun.security.") ) => // ok
+          case x: RuntimePermission  if( x.getName.startsWith("accessClassInPackage.sun.org.mozilla.javascript") ) => // ok
+          case x: RuntimePermission  if( x.getName.startsWith("accessClassInPackage.jdk.internal.org.objectweb.asm") ) => // ok
+          case x: RuntimePermission  if( runtimePerms.contains(x.getName)  ) => // ok
+          case x: ReflectPermission  if( reflectPerms.contains(x.getName)  ) => // ok
+          case x: PropertyPermission if( x.getActions == "read"            ) => // ok
+          case x: FilePermission if(x.getName.startsWith("/home/fanf") ) =>
```

Comment:
that was only for debugging, to allows println to be done just a few line below (else, we are stuck in a checkPermission loop). Remove it :)